core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -982,4 +983,13 @@ private String removeDotFromAMAndPM(SqlNode dateStringSqlNode) {
     }
     return dateString;
   }
+
+  @Override protected void unparseFormat(
+      final SqlWriter writer,
+      final SqlCall call, final int leftPrec, final int rightPrec) {
+    final SqlWriter.Frame formatFrame = writer.startFunCall("STRING");
```

Comment:
Also add a comment mentioning, why we have chosen "STRING()" as a compatible operand for "FORMAT()". The difference in the behavior of handling "null".

Reply:
Done